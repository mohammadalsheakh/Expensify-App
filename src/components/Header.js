import React from "react";
import { NavLink } from "react-router-dom"
const Header = () => (
    <div>
        <h1>Expensify</h1>
        < NavLink to="/" activeClassName="active" > Dash Bord </ NavLink>
        <br />
        < NavLink to="/create" activeClassName="active" > Create Expense Page </ NavLink>
        <br/>
        <NavLink to="/help"> Help </NavLink>
    </div>
)
export default Header