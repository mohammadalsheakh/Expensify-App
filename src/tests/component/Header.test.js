import Header from "../../components/Header";
import React from 'react';
import {shallow} from "enzyme"
// import toJson from "enzyme-to-json";
// import ReactShallowRenderer from 'react-test-renderer/shallow';

// test('should render Header correctly', () => {
//     const renderer = new ReactShallowRenderer();
//     renderer.render(<Header />);
//     expect(renderer.getRenderOutput()).toMatchSnapshot();
// });


test('should render Header correctly', () => {
    const wraper = shallow(<Header />) //here we pass the jsx we need to render or we need to test 
    expect(wraper).toMatchSnapshot()



    // expect(toJson(wraper)).toMatchSnapshot()
    // expect(wraper.find(`h1`).length).toBe(1)
    // expect(wraper.find(`h1`).text()).toBe(`Expensify`)
    
});
