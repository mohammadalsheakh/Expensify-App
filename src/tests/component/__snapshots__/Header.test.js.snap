// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="active"
    to="/"
  >
     Dash Bord 
  </NavLink>
  <br />
  <NavLink
    activeClassName="active"
    to="/create"
  >
     Create Expense Page 
  </NavLink>
  <br />
  <NavLink
    to="/help"
  >
     Help 
  </NavLink>
</div>
`;
